### PR TITLE
update Lottery logic

### PR DESCRIPTION
**Pull request by b0hdan1**

### CODE REVIEW

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 1]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public static Colors getRandomColor() {
+        int index = new Random().nextInt(Colors.values().length);
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: Random should be a class field
```suggestion
        int index = random.nextInt(Colors.values().length);
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Lottery.java:5]: Random and ColorSupplier should be class fields
```suggestion
public class Lottery {
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    public static String getRandomBall() {
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Lottery.java:6]: method should not be static
```suggestion
    public String getRandomBall() {
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+        Random random = new Random();
+        int value = random.nextInt(100);
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Lottery.java:8]: All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)

File: src/main/java/core/basesyntax/Colors.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public enum Colors {
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Colors.java:3]: naming convention: https://mate-academy.github.io/style-guides/java/java.html#s4.8.1-enum-classes
```suggestion
public enum Color {
```

File: src/main/java/core/basesyntax/Application.java [hunk 6]
```diff
@@ -1,7 +1,11 @@
 package core.basesyntax;
 
+import static core.basesyntax.Lottery.getRandomBall;
+
 public class Application {
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        System.out.println(getRandomBall());
+        System.out.println(getRandomBall());
+        System.out.println(getRandomBall());
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Application.java:9]: use loop

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Please, pay attention to common mistakes
https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery

File: src/main/java/core/basesyntax/ColorSupplier.java (hunk 1 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/ColorSupplier.java]: done

File: src/main/java/core/basesyntax/Lottery.java (hunk 2 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/Lottery.java]: done

File: src/main/java/core/basesyntax/Lottery.java (hunk 3 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/Lottery.java]: done

File: src/main/java/core/basesyntax/Lottery.java (hunk 4 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/Lottery.java]: done

File: src/main/java/core/basesyntax/Colors.java (hunk 5 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/Colors.java]: done

File: src/main/java/core/basesyntax/Application.java (hunk 6 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/Application.java]: done

File: src/main/java/core/basesyntax/Lottery.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int MAX_NUMBER = 100;
+    private final Random random = new Random();
+
+    public String getRandomBall() {
+        ColorSupplier colorSupplier = new ColorSupplier();
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/Lottery.java:10]: Please, make ColorSupplier class field

File: src/main/java/core/basesyntax/Lottery.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+        int value = random.nextInt(MAX_NUMBER);
+        Ball ball = new Ball(colorSupplier.getRandomColor(), value);
+        return ball.toString();
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/Lottery.java:13]: It's better to return `Ball` form this method, not `String`

File: src/main/java/core/basesyntax/Ball.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Ball {
+    private final String color;
+    private final int number;
+
+    public Ball(Color color, int number) {
+        this.color = String.valueOf(color);
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/Ball.java:8]: Please, [do not use String.valueOf](https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery#use-name-for-getting-string-representation-of-enum-constants) with enum values 
In this case its better to have constructor `public Ball(String color, int number)` and method `getRandomColor()` that returns String

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Lottery.java (hunk 7 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/Lottery.java:10]: done

File: src/main/java/core/basesyntax/Lottery.java (hunk 8 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/Lottery.java:13]: done

File: src/main/java/core/basesyntax/Ball.java (hunk 9 above)
Comment (b0hdan1) [reply src/main/java/core/basesyntax/Ball.java]: done

**Review by Ivan95kos — APPROVED** [review]
Great job

**Pull request closed without merging**